sklearn/linear_model/ransac.py

```diff
@@ -243,6 +247,17 @@ def fit(self, X, y):
         except ValueError:
             pass
 
+        estimator_fit_has_sample_weight = has_fit_parameter(base_estimator, "sample_weight")
+        estimator_name = type(base_estimator).__name__
+        if (sample_weight is not None
+                    and not estimator_fit_has_sample_weight ):
+            raise TypeError("%s does not support sample_weight. Samples"
```

Comment:
It seems to me that a ValueError would be more appropriate. `sample_weight` is still of the expected type